work/v1/types.go

```diff
@@ -225,3 +225,68 @@ type ManifestWorkList struct {
 	// Items is a list of manifestworks.
 	Items []ManifestWork `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AppliedManifestWork represents an applied manifestwork on managed cluster. It is placed
+// on managed cluster. An AppliedManifestWork links to a manifestwork on a hub recording resources
+// deployed in the managed cluster.
+// When the agent is removed from managed cluster, cluster-admin on managed cluster
+// can delete appliedmanifestwork to remove resources deployed by the agent.
+type AppliedManifestWork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired configuration of AppliedManifestWork
+	Spec AppliedManifestWorkSpec `json:"spec,omitempty"`
+
+	// Status represents the current status of AppliedManifestWork
+	// +optional
+	Status AppliedManifestWorkStatus `json:"status,omitempty"`
+}
+
+// AppliedManifestWorkSpec represents the desired configuration of AppliedManifestWork
+type AppliedManifestWorkSpec struct {
+	// HubID represents the ID of the hub that the agent fetches ManifestWork. It could
+	// be a hash of hub kubernetes apiserver url.
+	// +required
+	HubID string `json:"hubID"`
+
+	// ManifestWorkName represents the name of the related manifestwork on hub.
+	// +required
+	ManifestWorkName string `json:"manifestWorkName"`
+
+	// ClusterName represents the name of the cluster on hub where the related manifestwork
+	// is placed.
+	// +required
+	ClusterName string `json:"clusterNamee"`
```

Comment:
this seems unnecessary since it will be the same for every instance in the cluster.

Reply:
I am not sure if there is a case that there are multiple agents on one spoke that register to a single hub?

Reply:
> I am not sure if there is a case that there are multiple agents on one spoke that register to a single hub?

I don't think that will behave properly at an API level since you cannot predict which will service you.  Let's start without it.